src/main/resources/META-INF/plugin.xml

```diff
@@ -430,7 +431,35 @@
                 implementation="com.redhat.devtools.lsp4ij.features.semanticTokens.LSPSemanticTokensHighlightVisitor"/>
         <colorSettingsPage
                 implementation="com.redhat.devtools.lsp4ij.features.semanticTokens.SemanticTokensColorSettingsPage"/>
+    </extensions>
 
+    <!-- Semantic tokens-based file view provider -->
+    <extensions defaultExtensionNs="com.intellij">
+        <!-- File view provider based on textDocument/semanticTokens -->
+        <!--
+         NOTE: This can only be registered for TextMate files because abstract file type plain text files have no
+         language and a variable file type.
+         -->
+        <lang.fileViewProviderFactory
+                language="textmate"
+                implementationClass="com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokensFileViewProviderFactory"/>
+        <lang.elementManipulator
```

Comment:
It's safe to have these EP registrations here as they're language-/file type-independent.